packages/service-library/src/servicelib/archiving_utils.py

```diff
@@ -141,49 +161,81 @@ async def unarchive_dir(
-                try:
-                    file_name_in_archive = (
-                        _strip_directory_from_path(file_to_add, dir_to_compress)
-                        if store_relative_path
-                        else file_to_add
-                    )
-
-                    # because surrogates are not allowed in zip files,
-                    # replacing them will ensure errors will not happen.
-                    escaped_file_name_in_archive = _strip_undecodable_in_path(
-                        file_name_in_archive
-                    )
-
-                    zip_file_handler.write(file_to_add, escaped_file_name_in_archive)
-                except ValueError as value_error:
-                    log.exception("Could write files to archive, please check logs")
-                    raise value_error
+            desc = f"compressing {dir_to_compress} -> {destination}"
+            with tqdm(
+                desc=f"{desc}\n", total=folder_size_bytes, **_TQDM_FILE_OPTIONS
+            ) as pbar:
+                for file_to_add in _iter_files_to_compress(
+                    dir_to_compress, exclude_patterns
+                ):
+                    pbar.set_description(f"{desc}:{file_to_add.name}")
+                    try:
+                        file_name_in_archive = (
+                            _strip_directory_from_path(file_to_add, dir_to_compress)
+                            if store_relative_path
+                            else file_to_add
+                        )
+
+                        # because surrogates are not allowed in zip files,
+                        # replacing them will ensure errors will not happen.
+                        escaped_file_name_in_archive = _strip_undecodable_in_path(
+                            file_name_in_archive
+                        )
+
+                        def _write_with_progress(
+                            zip_write_fct, self, data  # pylint: disable=unused-argument
+                        ):
+                            pbar.update(len(data))
+                            return zip_write_fct(data)
+
+                        # Replace original write() with a wrapper to track progress
+                        assert zip_file_handler.fp  # nosec
+                        zip_file_handler.fp.write = types.MethodType(
+                            partial(
+                                _write_with_progress,
+                                zip_file_handler.fp.write,
+                            ),
+                            zip_file_handler.fp,
+                        )
+                        # zip_file_handler.write(in_file)
+                        zip_file_handler.write(
+                            file_to_add, escaped_file_name_in_archive
+                        )
+                    except ValueError as value_error:
+                        log.exception("Could write files to archive, please check logs")
```

Comment:
if it is reraised, why to log the exception. Shouldn't you expect the caller to deal with that? what if the caller also logs the exception?

Reply:
hmm yep. will change that

Reply:
I also vote to drop it

Reply:
don't why it was like that.. simplified.